lib/modules.nix

```diff
@@ -87,6 +84,15 @@ rec {
       result = { inherit options config; };
     in result;
 
+
+ /* Remove modules if there is another module that 'disables' it it. */
```

Comment:
Maybe use something like this: "Return the non-disabled modules. Disabling modules allows replacing their implementations."